Sources/Datadog/Logger.swift

```diff
@@ -401,9 +401,9 @@ public class Logger {
         private func buildOrThrow() throws -> Logger {
             guard let loggingFeature = LoggingFeature.instance else {
                 throw ProgrammerError(
-                    description: Datadog.instance == nil
-                        ? "`Datadog.initialize()` must be called prior to `Logger.builder.build()`."
-                        : "`Logger.builder.build()` produces a non-functional logger, as the logging feature is disabled."
+                    description: Datadog.isInitialized
```

Comment:
Sooner or later we'll need to also remove `Datadog` references like this and change them to use `DatadogCore`. Now we could replace this with the use of `DefaultDatadogCore`. Is the plan to change things like this in separate PR(s) or do you think it's better to wait and tackle it at once when isolating real modules?

Reply:
Indeed, this will be removed when we remove each feature singleton. Once all features are migrated, we will rely on injected core.